tests/helpers/execute.ts

```diff
@@ -33,13 +42,28 @@ interface ExecuteReturn {
 
 export const executeCli = async (args: string[] = [], opts: { env: Env } = { env: undefined }): Promise<ExecuteReturn> => {
   const { env = undefined } = opts;
-  const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
-  childProcess.stdin.setDefaultEncoding('utf-8');
   const promise = new Promise<ExecuteReturn>((resolve, reject) => {
+    const childProcess = createProcess('./src/index.ts', ['--progress=false', ...args], env);
+    childProcess.stdin.setDefaultEncoding('utf-8');
+    let stdout!: string, stderr!: string;
+    const stdoutPromise = readStream(childProcess.stdout).then((value: undefined | string) => (stdout = value ?? ''));
+    const stderrPromise = readStream(childProcess.stderr).then((value: undefined | string) => (stderr = value ?? ''));
     childProcess.once('exit', (code) => {
-      Promise.all([readStream(childProcess.stdout), readStream(childProcess.stderr)])
-        .then(([stdout, stderr]) => resolve({ exitCode: code, stdout, stderr }))
-        .catch(reject);
+      stderrPromise
+        .catch(() => {
+          throw new Error('read stderr Stream failed');
+        })
+        .then(() => {
+          stdoutPromise
+            .catch(() => {
+              throw new Error('read stdout Stream failed');
+            })
+            .then(() => {
+              resolve({ exitCode: code, stdout, stderr });
+            })
+            .catch(() => 'obligatory catch');
+        })
+        .catch(() => 'obligatory catch');
```

Comment:
the nesting level is too deep, either deconstruct it to smaller promises and functions

Reply:
fixed